gc3libs/utils.py

```diff
@@ -33,6 +33,7 @@
 
 
 from collections import defaultdict, deque
+from functools import total_ordering
```

Comment:
`total_ordering` is not available in Py2.6, which we still need to support.  I agree it would simplify code here, can you please add a backport somewhere in `gc3libs.compat` and use that if `functools` does not provide it?

Reply:
oh damn, how annoying.

actually i did some more thinking about this. total_ordering is just used on the `PlusInfinity` object, which itself isn't used all that often, and is kind of hacky. It is used to ensure that more/less than comparisons with some known object will return True/False.

Why not remove total_ordering, remove PlusInfinity, and simply change the few instances of:

`boolean = x > PlusInfinity()`

to

`boolean = False`?

I feel like I'm missing something somewhere, because I can't see why the Singleton and these classes need to exist in the first place?

Reply:
Also, we could just use: https://pypi.org/project/total-ordering/

Reply:
> I can't see why [...] these classes need to exist in the first place?

They're used as a shortcut: e.g., in the `Engine` class (in `core.py`) we have couple of limits (`limit_submitted`); now, there are cases when you want to *not* have a limit; to support those, you need to add a special "unlimited" value (e.g. `limit_submitted=None`) or an additional flag (`do_limit_submitted=True`) and then make sure that every `if` involving this limit checks whether the limit should be checked:

```python
if limit_submitted is None or num_submitted < limit_submitted:
  # do submit some more
```

Or you just set `limit_submitted=PlusInfinity()` and code like you always have a limit. :-)

Reply:
OK, I understand the point now, but I'm really not convinced that the addition of this object simplifies things overall. I would much rather be explicit in `core.py` with a simple additional comparison of `if limit is None`, rather than comparing against this kind of counterintuitive object. Seems like a few added comparisons (which are semantically very clear about what they mean, i.e. "limit not set") is much simpler than custom comparison objects.

Reply:
I have quite the opposite view here: we've gone that route for the "requirements" objects (memory limits etc) and it's been a constant source of bugs (somebody adds a new constraint on the requirements but forgets to add the check that it is not `None`)... I really regretted it and would rather change the requirements spec to use the same Plus/MinusInfinity interface.

Reply:
OK, fine to keep PlusInfinity if you've found it helpful. But MinusInfinity is never used, and I see no compelling reason to make them singletons. So todo is just set up total_ordering backport

Reply:
Ok!